### PR TITLE
Case sensitive html parser

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: modules/angular2/src/compiler/html_tags.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import {isPresent, isBlank, normalizeBool, CONST_EXPR} from 'angular2/src/facade/lang';
+
+// TODO: fill this!
```
Comment (vicb) [inline modules/angular2/src/compiler/html_tags.ts]: remove todo
by purpose not complete....

File: modules/angular2/src/compiler/html_tags.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+           (isBlank(currentParent) || this.requiredParent != currentParent.toLowerCase());
+  }
+
+  isClosedByChild(name: string) {
+    return normalizeBool(this.closedByChildren['*']) ||
+           normalizeBool(this.closedByChildren[name.toLowerCase()]);
+  }
+}
+
+// TODO: Fill this table using
```
Comment (vicb) [inline modules/angular2/src/compiler/html_tags.ts]: simplified version of the spec...

File: modules/angular2/test/compiler/html_parser_spec.ts [hunk 3]
```diff
@@ -18,131 +19,249 @@ import {
...
+          expect(humanizeDom(parser.parse('<svg></svg>', 'TestComp')))
+              .toEqual([[HtmlElementAst, '@svg:svg', 0]]);
+        });
+
+        it('should propagate the namespace', () => {
+          expect(humanizeDom(parser.parse('<myns:div><p></p></myns:div>', 'TestComp')))
+              .toEqual([[HtmlElementAst, '@myns:div', 0], [HtmlElementAst, '@myns:p', 1]]);
+        });
+
+        it('should lowercase tag names', () => {
```
Comment (vicb) [inline modules/angular2/test/compiler/html_parser_spec.ts]: update the desc

**Comment by vicb** [comment]
Reviewed in person with @tbosch

**Comment by vicb** [comment]
TODO: ref https://github.com/angular/angular/issues/4417

**Comment by tbosch** [comment]
Error messages: take previous and following 2 lines, instead of -1- and +10 chars

**Comment by tbosch** [comment]
Entities: lt, gt, nbsp

**Pull request closed without merging**